src/Filter.Nest/FilterLogic.cs

```diff
@@ -39,10 +39,104 @@ internal static class FilterLogic
                     var queries = new List<Func<QueryContainerDescriptor<T>, QueryContainer>>();
                     var aliasAttribute = validValueProperty.GetCustomAttribute<MappingAliasAttribute>();
 
-                    if (typeof(Range.Generic.IRange<>).IsAssignableFrom(filterProperty.PropertyType))
+                    if (filterPropertyValue.GetType().GetInterfaces().Any(
+                        x => x.IsGenericType && x.GetGenericTypeDefinition() == typeof(Range.Generic.IRange<>)))
                     {
-                        
-                        
+                        Range.Generic.Range<DateTimeOffset> dateRange;
+                        Range.Generic.Range<decimal> numericRange;
+
+                        if ((dateRange = Range.Range.AsDateRange(filterPropertyValue)) != null)
+                        {
+                            queries.Add(x =>
+                                x.DateRange(y =>
+                                {
+                                    if (dateRange.MinValue.HasValue)
+                                    {
+                                        var minValue = dateRange.MinValue.Value.Date;
+
+                                        if (aliasAttribute != null)
+                                        {
+                                            y = dateRange.IsMinInclusive
+                                                ? y.Field(aliasAttribute.Alias).GreaterThanOrEquals(minValue)
```

Comment:
I tried to minimize this branching logic related to setting `Field`:

```csharp
var fieldName = aliasAttribute != null
    ? (dynamic)aliasAttribute.Name
    : validValueProperty;
```

While this compiles, it also results in a `StackOverflowException`–cause unknown.